src/main/java/org/springframework/data/couchbase/core/ReactiveTemplateSupport.java

```diff
@@ -20,11 +20,15 @@
 import org.springframework.data.couchbase.core.mapping.CouchbaseDocument;
 import org.springframework.data.couchbase.core.mapping.event.CouchbaseMappingEvent;
 
+/**
+ *
+ * @author Michael Reiche
+ */
 public interface ReactiveTemplateSupport {
 
 	Mono<CouchbaseDocument> encodeEntity(Object entityToEncode);
 
-	<T> Mono<T> decodeEntity(String id, String source, long cas, Class<T> entityClass);
+	<T> Mono<T> decodeEntity(String id, String source, long cas, Class<T> entityClass, String scope, String collection);
```

Comment:
I assume our template support classes are not public API so we can break them by extending?

Reply:
Good point. They are public. So I'll keep make a method with the old signature to call this one.